presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleConfig.java

```diff
@@ -68,17 +67,4 @@ public OracleConfig setNumberRoundingMode(RoundingMode numberRoundingMode)
         this.numberRoundingMode = numberRoundingMode;
         return this;
     }
-
-    @Min(4000)
-    public int getVarcharMaxSize()
```

Comment:
starting on oracle 12c varchar could be increased from 4000 to 32767 bytes (any value between) using [Extended Data Types](https://oracle-base.com/articles/12c/extended-data-types-12cR1#bugs)

Reply:
I reintroduced the config parameter. Just changed the name slightly, so it is obvious that limit is about number of bytes. Not characters.

Reply:
could be added the  @Max(32767)

Reply:
Actually can it be something in between 4000 and 32767. Or just 4000 or 32767?
If latter is the case maybe we should make the config parameter `oracle.enable-extended-datatypes=true/false`?

Reply:
@losipiuk the true/false may work, checking the [documentation](https://docs.oracle.com/database/121/REFRN/GUID-D424D23B-0933-425F-BC69-9C0E6724693C.htm#REFRN10321) to activate that is used
ALTER SYSTEM SET max_string_size=extended SCOPE=SPFILE;

that change this values:
    VARCHAR2 : 4000 bytes
    NVARCHAR2 : 4000 bytes
    RAW : 2000 bytes

to 
    VARCHAR2 : 32767 bytes
    NVARCHAR2 : 32767 bytes
    RAW : 32767 bytes

Reply:
Why exactly do we have this setting? What would be the downside to having this hard-coded, at 4000 (safe value)?
(since this isn't obvious, we should capture the reasoning in the code comment at least)

If desired, in the future, we could try to detect server configuration and adapt, without user needing to set anything.

Reply:
I kinda agree with what @findepi wrote. @eskabetxe do you feel strongly about the functionality and can provide rationale? If not I would simplify configuration (after all, the lest configuration switches the better) and live with 4000 for now.

Reply:
I updated the PR removing `oracle.varchar2.max-bytes`. We may discuss later on if we want to add it (in this or other form) as a followup.

Reply:
@findepi  this allow you to use the extended data types added on 12c
you can use the hard-coded value, but if the database have the extended version activated and on creation for example you create a table with varchar (5000), the connector will create a nlob field (because 5000 > than 4000 limit)

Reply:
I think we all understand what having a parameter changes in logic for `CREATE TABLE` issued from Presto to create Oracle table. The question is, is the benefit worth added complexity and extra configuration parameter. As @findepi mentioned above it can be improved to support higher limits without explicit configuration parameter. Also I believe that typical usecase for Oracle connector is to query tables created in Oracle already. Not to create new ones.